test/jdk/java/nio/channels/DatagramChannel/SelectWhenRefused.java

```diff
@@ -70,12 +70,13 @@ public static void main(String[] args) throws IOException {
                     DatagramChannel.open().bind(refuser).close();
                     throw new RuntimeException("Unexpected wakeup");
                 }
-
+            }
```

Comment:
This should be:
```
                    if (ignoreStrayWakeup) {
                        if (i < MAX_TRIES - 1) continue;
                    }
                    // BindException will be thrown if another service is using 
                    // our expected refuser port, cannot run just exit.
                    DatagramChannel.open().bind(refuser).close();
                    throw new RuntimeException("Unexpected wakeup");
                } else {
                    break; 
                }
```
We don't want to silently finish and exit the loop if the last attempt failed. And we don't want to retry if n == 0.